packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt

```diff
@@ -0,0 +1,137 @@
+import com.amplifyframework.core.model.query.Where
+import com.amplifyframework.core.model.query.predicate.QueryField.field
+import com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.not
+import com.amplifyframework.core.model.temporal.Temporal
+import com.amplifyframework.datastore.AWSDataStorePlugin
+import com.amplifyframework.datastore.DataStoreCategory
+import com.amplifyframework.datastore.DataStoreException
+import com.amplifyframework.datastore.appsync.SerializedModel
+import io.flutter.plugin.common.MethodChannel
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.invocation.InvocationOnMock
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class AmplifyDataStorePluginTest {
+    lateinit var plugin: AmplifyDataStorePlugin
+    private var mockDataStore = mock(DataStoreCategory::class.java)
+    private var mockAmplifyDataStorePlugin = mock(AWSDataStorePlugin::class.java);
+    private val mockResult: MethodChannel.Result = mock(MethodChannel.Result::class.java)
+
+    private val amplifySuccessResults = mutableListOf<SerializedModel>(
+            SerializedModel.builder()
+                    .serializedData(
+                            mapOf("id" to "4281dfba-96c8-4a38-9a8e-35c7e893ea47",
+                                  "title" to "Title 1",
+                                  "rating" to 4))
+                    .modelName("Post")
+                    .id("4281dfba-96c8-4a38-9a8e-35c7e893ea47")
+                    .build(),
+            SerializedModel.builder()
+                    .serializedData(
+                            mapOf("id" to "43036c6b-8044-4309-bddc-262b6c686026",
+                                  "title" to "Title 2",
+                                  "created" to Temporal.DateTime("2020-02-20T20:20:20-08:00")))
+                    .modelName("Post")
+                    .id("43036c6b-8044-4309-bddc-262b6c686026")
+                    .build()
+    )
+
+    @Before
+    fun setup() {
+        plugin = AmplifyDataStorePlugin()
+        setFinalStatic(Amplify::class.java.getDeclaredField("DataStore"), mockDataStore)
+        `when`(mockDataStore.getPlugin("awsDataStorePlugin")).thenReturn(mockAmplifyDataStorePlugin)
+    }
+
+    @Test
+    fun test_Query_Success_Result() {
+        doAnswer { invocation: InvocationOnMock ->
+            assertEquals(invocation.arguments[0], "Post")
+            assertEquals(invocation.arguments[1], Where.matchesAll())
```

Comment:
Instead of having verifications inside an `Answer` closure, I would suggest using a `verify` with an `ArgumentCaptor`. Then, make assertions on the captured value.

```java
ArgumentCaptor<Gizmo> captor = ArgumentCaptor.forClass(Gizmo.class);
verify(parent).someMethod(captor.capture());
Gizmo actualGizmo = captor.value();
assertEquals(exepctedGizmo, actualGizmo);
```

Same for the other `Answer` block(s), too.